src/main/java/org/firmata4j/firmata/transport/AbstractFirmataTransport.java

```diff
@@ -0,0 +1,32 @@
+package org.firmata4j.firmata.transport;
+
+import org.firmata4j.firmata.FirmataParser;
+
+import java.io.IOException;
+
+/**
+ * thew hardware than Firmata is implemented on
+ */
+public abstract class AbstractFirmataTransport implements FirmataTransportInterface {
+
+    private FirmataParser parser;
+
+    protected final void queueDeviceResponse(byte[] bytes) {
+        parser.parse(bytes);
+    }
+
+    @Override
+    public final void startTransport() throws IOException {
+        parser = new FirmataParser();
```

Comment:
`parser` should not be instantiated by a transport but rather provided to the trasport from the outside because transport should not pick the protocol or its version. So it makes `AbstractFirmataTransport` redundant because `FirmataTransportInterface` will be sufficient. And it makes sense because there is nothing comon between two transports (serial and TCP/IP for example).